### ISSUE TITLE
Button list

### ISSUE THREAD

**Issue by rayman89** [M-Reimer/undoclosetab#25]
Could you add an extra button or preference so that when you click the button it shows the list of closed tabs? That way you could put as many closed tabs as you want. Also would be nice to have favicons and include closed windows as well.

**Comment by M-Reimer** [comment]
**List of closed tabs on click**
There are other addons on addons.mozilla.org which do exactly this. There is no need to clone something that is already there. For example:
https://addons.mozilla.org/de/firefox/addon/undo-close-tab-button/

My "Undo Close Tab" is meant to have both. The list and the "single click restore". If we want to have both, we are limited by the API and at least so far there is no way around this.

**Favicons**
Are already available, but you need at least Firefox 55 to see them.
The screenshot on addons.mozilla.org is outdated. I'll add a new one as soon as I find the time.

**Closed windows**
I've added this to the following issue: https://github.com/M-Reimer/undoclosetab/issues/17
If I ever get this API, then "shift + click" to reopen the last closed tab would be possible.

**Comment by M-Reimer** [comment]
Forgot to add: If you want a menu without the "More items" submenu, then you always have the option to enable the additional context menus in the addon configuration. The menus in page or tab context menu don't need this additional menu level, so you have a full list.

**Comment by rayman89** [comment]
Sorry If my request bothered you or made you angry. What I mean by restore the window is for example if I open a new window with a facebook tab or something when I want to re open it I'm not able to do so. The facebook tab does not appear in the list (and yes the checkbox to only show current window closed tab is checked off).

**Comment by M-Reimer** [comment]
No, your request didn't bother me or made me angry. English is not my native language, so there is a chance that things come off the wrong way. Sorry for that.

You are right. Closed windows aren't handled in any way, so far. And it would be pretty difficult to do it directly. Windows don't have titles or favicons. But it should be possible to handle the tabs that were in closed windows. Of course only if "only show current..." is unchecked. I don't know what happens if I try to reopen tabs of closed windows, but I'll try and if this works, I'll add this as a fix for the already existent "list tabs from all windows" feature.

**Comment by M-Reimer** [comment]
This is way more difficult to fix as I initially thought.
With the existing API it seems to be not possible, but I'll have to investigate further and create a small test addon to send to Mozilla.
Seems like I have to create an upstream bug for this as soon as I have the test addon.

What I would like to have is basically that the list of closed tabs of a closed windows keeps visible in the menu and the closed window is restored if one of the tabs has to be restored.
So far I have found no way to get the list of tabs of a closed window.
And even adding the window as one single tab is not possible as I was not able to get the window title.
